### PR TITLE
cache: fix LRU, expose resident properties

### PR DESCRIPTION
**Pull request by witchel**
Before #2540

### CODE REVIEW

File: scripts/setup_dev_ubuntu.sh [hunk 1]
```diff
@@ -31,8 +31,8 @@ sudo bash -x "${REPO_ROOT}/scripts/setup_ubuntu.sh" --no-setup-toolchain-variant
 "${REPO_ROOT}/scripts/setup_conan.sh"
 
 # make clang-{tidy,format}-12 the default
-sudo update-alternatives --verbose --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-12 90
-sudo update-alternatives --verbose --install /usr/bin/clang-format clang-format /usr/bin/clang-format-12 90
+sudo update-alternatives --verbose --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-12 100
+sudo update-alternatives --verbose --install /usr/bin/clang-format clang-format /usr/bin/clang-format-12 100
```
Comment (roshandathathri) [inline scripts/setup_dev_ubuntu.sh:35]: Why this change?

**Review by roshandathathri — APPROVED** [review]
LGTM

File: scripts/setup_dev_ubuntu.sh (hunk 1 above)
Comment (witchel) [reply scripts/setup_dev_ubuntu.sh:35]: Good question.

Because when I ran those lines on our servers with a weight of 90, they left clang-format as version 10.  Only by increasing the number did it actually update them to version 12.
